mempool/src/pool/tests/mod.rs

```diff
@@ -122,6 +130,52 @@ async fn add_single_tx() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn add_tx_with_fee_rate_below_minimum() {
+    let min_relay_fee_rate = FeeRate::from_amount_per_kb(Amount::from_atoms(123));
+    let mut mempool = setup_with_min_tx_relay_fee_rate(min_relay_fee_rate);
+
+    async fn make_tx(
+        mempool: &Mempool<StoreMemoryUsageEstimator>,
+        relay_fee: Fee,
+    ) -> SignedTransaction {
+        let outpoint_source_id = mempool.chain_config.genesis_block_id().into();
+        let flags = 0;
+        let input = TxInput::from_utxo(outpoint_source_id, 0);
+
+        tx_spend_input(
+            mempool,
+            input,
+            InputWitness::NoSignature(Some(DUMMY_WITNESS_MSG.to_vec())),
+            relay_fee,
+            flags,
+        )
+        .await
+        .unwrap()
+    }
+
+    let estimated_tx_size = make_tx(&mempool, Amount::ZERO.into()).await.encoded_size();
+    let min_relay_fee = min_relay_fee_rate.compute_fee(estimated_tx_size).unwrap();
+
+    // Tx1's fee is below the minimum, so it must be rejected.
+    let tx1_relay_fee = (min_relay_fee - Amount::from_atoms(1).into()).unwrap();
+    let tx1 = make_tx(&mempool, tx1_relay_fee).await;
+
+    let err = mempool.add_transaction_test(tx1).unwrap_err();
+    assert!(matches!(
+        err,
+        Error::Policy(MempoolPolicyError::InsufficientFeesToRelay {
+            tx_fee: _,
+            min_relay_fee: _
+        })
+    ));
```

Comment:
You can also assert ban score of the error is 0 here.

Reply:
Won't it be redundant though? The corresponding p2p test already checks that the peer won't be punishes in such a situation.
Also, should mempool tests care about ban scores at all? As I can see, the existing ones don't.

Reply:
Yea it probably is redundant. Was thinking that if something can be checked using simpler setup, it should be preferred. Checking the ban score here only uses mempool (which is sufficient, everything needed to check it is defined in the mempool crate) rather than pulling in mempool _and_ p2p. Just a small suggestion though, doing it in p2p is fine too.

Reply:
The corresponding test in p2p is not really optional, because it actually checks the outcome of the transaction being received from a peer, rather than a ban score of certain error, which may or may not be taken into account later.

Then, checking the ban score here IMO is not only redundant, but it's also a bit strange. I.e. I've already checked here that a certain error is produced, so checking its ban score is equivalent to writing a separate test that only checks the ban score of that specific error. So the question arises why we only check the ban score of that error and not other ones.

Also, I don't think that ban scores, being a p2p concept, should exist outside of p2p at all. I think that the corresponding trait and its implementations should be moved to p2p eventually.

So, since you don't insist, I'm leaving this as is.